tests/APIv2/MessagesTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * @author Alexandre (DaazKu) Chouinard <alexandre.c@vanillaforums.com>
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+namespace VanillaTests\APIv2;
+
+/**
+ * Test the /api/v2/messages endpoints.
+ */
+class MessagesTest extends AbstractResourceTest {
+
+    private static $userID;
+
+    private static $conversationID;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($name = null, array $data = [], $dataName = '') {
+        $this->baseUrl = '/messages';
+
+        parent::__construct($name, $data, $dataName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function setUpBeforeClass() {
+        parent::setupBeforeClass();
+
+        /** @var \UsersApiController $usersAPIController */
+        $usersAPIController = static::container()->get('UsersAPIController');
+
+        $user = $usersAPIController->post([
+            'name' => "MessagesUser1",
+            'email' => "MessagesUser1@example.com",
+            'password' => "$%#$&ADSFBNYI*&WBV1",
+        ]);
+        self::$userID = $user['userID'];
+
+        /** @var \ConversationsApiController $conversationsAPiController */
+        $conversationsAPiController = static::container()->get('ConversationsAPiController');
+
+        $conversation = $conversationsAPiController->post([
+            'participantIDs' => [self::$userID]
+        ]);
+        self::$conversationID = $conversation['conversationID'];
+
+        // Disable email sending.
+        /** @var \Gdn_Configuration $config */
+        $config = static::container()->get('Config');
+        $config->set('Garden.Email.Disabled', true, true, false);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function record() {
+        return array_merge(parent::record(), ['conversationID' => self::$conversationID]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function indexUrl() {
+        $indexUrl = $this->baseUrl;
+        $indexUrl .= '?'.http_build_query(['conversationID' => self::$conversationID]);
+        return $indexUrl;
+    }
+
+ /**
+     * {@inheritdoc}
+     */
+    public function testDelete() {
```

Comment:
PHPUnit supports [skipping tests with a `@requires` tag](https://phpunit.de/manual/5.7/en/incomplete-and-skipped-tests.html#incomplete-and-skipped-tests.skipping-tests-using-requires). You could add `@requires MessagesApiController::delete` here and it'll skip this test until that function is implemented in the API. You make the "test" a simple call to `fail` that states the unit test needs to be added. It'll make sure adding the test isn't forgotten when DELETE, etc are implemented.

Reply:
How is that different from marking the test as skipped with a clear message like I did right below?

Reply:
Because when DELETE is added to the controller, the test will fail (and so will the build) if you haven't removed the placeholder. Using your method, when you (or whoever) adds DELETE, they'll need to remember to add the unit test, because it'll continue to just skip over it.

Reply:
I totally misread that. That's way better actually.